psi4/src/psi4/lib3index/df_helper.cc

```diff
@@ -182,71 +191,89 @@ void DF_Helper::initialize()
-                        {
-                            fun_prints[omu*nao_+onu]= val;
-                            fun_prints[onu*nao_+omu]= val;
-                        }
-                    }
+    std::shared_ptr<TwoBodyAOInt> eri = std::shared_ptr<TwoBodyAOInt>(schwarzfactory.eri());
+    const double* buffer = eri->buffer();
+
+//    double val, max_val = 0.0;
+//    size_t MU, NU, mu, nu, omu, onu, nummu, numnu, index;
+//    #pragma omp parallel for private(MU, NU, mu, nu, omu, onu, nummu, numnu, index, val, rank) num_threads(nthreads_) schedule(guided)
+//    for (MU=0; MU < pshells_; ++MU) {
+//#ifdef _OPENMP
+//        rank = omp_get_thread_num();
+//#endif
+//        nummu = primary_->shell(MU).nfunction();
+//        for (NU=0; NU <= MU; ++NU) {
+//            numnu = primary_->shell(NU).nfunction();
+//            eri[rank]->compute_shell(MU,NU,MU,NU);
+//            for (mu=0; mu < nummu; ++mu) {
+//                omu = primary_->shell(MU).function_index() + mu;
+//                for (nu=0; nu < numnu; ++nu) {
+//                    onu = primary_->shell(NU).function_index() + nu;
+//                    if (omu>=onu) {
+//                        index = mu*(numnu*nummu*numnu+numnu)+nu*(nummu*numnu+1);
+//                        val = fabs(buffer[rank][index]);
+//                        max_val = (max_val < val ? val : max_val);
+//                        if (shell_prints[MU*pshells_+NU] <= val){
+//                            shell_prints[MU*pshells_+NU]=val;
+//                            shell_prints[NU*pshells_+MU]=val;
+//                        }
+//                        if (fun_prints[omu*nao_+onu] <= val)
+//                        {
+//                            fun_prints[omu*nao_+onu]= val;
+//                            fun_prints[onu*nao_+omu]= val;
+//                        }
+//                    }
+//                }
+//            }
+//        }
+//    }
+//  tolerance_ = cutoff_*cutoff_/max_val;
+    double max_ = 0.0;
+    for (int P = 0; P < pshells_; P++) {
+        for (int Q = 0; Q <= P; Q++) {
+            int nP = primary_->shell(P).nfunction();
+            int nQ = primary_->shell(Q).nfunction();
+            int oP = primary_->shell(P).function_index();
+            int oQ = primary_->shell(Q).function_index();
+            eri->compute_shell(P, Q, P, Q); 
+            double max_val = 0.0;
+            for (int p = 0; p < nP; p++) {
+                for (int q = 0; q < nQ; q++) {
+                    max_val = std::max(max_val, fabs(buffer[p * (nQ * nP * nQ + nQ) + q * (nP * nQ + 1)]));
+                }
+            }
+            max_ = std::max(max_, max_val);
+            shell_prints[P * pshells_ + Q] = shell_prints[Q * pshells_ + P] = max_val;
+            for (int p = 0; p < nP; p++) {
+                for (int q = 0; q < nQ; q++) {
+                    fun_prints[(p + oP) * nao_ + (q + oQ)] =
+                    fun_prints[(q + oQ) * nao_ + (p + oP)] = max_val;
```

Comment:
Since this only scans through a 2-dimensional quantity (nao*nao), threading slows things down for smaller systems and does not boost much for larger ones.  I can revert it back and show you some timings.